src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,53 +1,114 @@
-import { useState } from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [formValues, setFormValues] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const isButtonDisabled = () => {
+    const {
+      title,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = formValues;
+
+    return !(title.trim() && imgUrl.trim()
+      && imdbUrl.trim() && imdbId.trim());
+  };
+
+  const handleChange = (value: string, name: string) => {
+    setFormValues((prevState) => ({
```

Comment:
You could take field name from event target 

```suggestion
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
  const { value, name } = e.target
    setFormValues((prevState) => ({
```